src/main/java/com/blackduck/integration/sca/upload/file/FileUploaderResponse.java

```diff
@@ -0,0 +1,109 @@
+package com.blackduck.integration.sca.upload.file;
+
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.Map;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+
+import com.blackduck.integration.rest.RestConstants;
+import com.blackduck.integration.rest.exception.IntegrationRestException;
+import com.blackduck.integration.rest.response.Response;
+
+public class FileUploaderResponse implements Response {
+    private final IntegrationRestException integrationRestException;
+
+    public FileUploaderResponse(IntegrationRestException integrationRestException) {
+        this.integrationRestException = integrationRestException;
+    }
+
+    @Override
+    public HttpUriRequest getRequest() {
+        throw new NotImplementedException();
```

Comment:
Rather than NotImplementedExceptions, can we switch to UnsupportedOperationException here and in the other cases within this class?
If we have no intention to implement these methods so I think it's better to just declare they aren't supported at all. Plus we get the added benefit of the exception class coming from java native rather than a dependency on Apache.

Reply:
We should use the UnsupportedOperationException as mentioned here since it is native to Java.